File: 이상욱/BOJ/m3w10/BOJ1548.java
package m2w10;

import java.io.*;
import java.util.*;

public class BOJ1548{
    public static void main(String[] args) throws Exception{
        
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;
        
        int n = Integer.parseInt(br.readLine());
        int[] arr = new int[n];
        
        st = new StringTokenizer(br.readLine());
        
        for(int i = 0; i < n; i++){
            arr[i] = Integer.parseInt(st.nextToken());
        }
        
        if(arr.length <= 2){
            System.out.println(arr.length);
            return;
        }
        
        Arrays.sort(arr);
        
        int max = -1;
        for(int i = 0; i < n-2; i++){
            for(int j = n-1; j > i+1; j--){
                if(arr[i] + arr[i+1] > arr[j]){
                    max = Math.max(max,j-i+1);
                }
            }
        }
        
        if(max == -1){
            max = 2;
        }
        
        System.out.println(max);
    }
}

File: 이주은/BOJ_Kotlin/10주차/BOJ_1548.kt
package `10주차`

import java.io.BufferedReader
import java.io.InputStreamReader
import java.util.*

fun main()= with(BufferedReader(InputStreamReader(System.`in`))){
    val n = readLine().toInt()

    val arr = IntArray(n)
    var st = StringTokenizer(readLine())

    for(i in 0 until n){
        arr[i] = st.nextToken().toInt()
    }

    Arrays.sort(arr)
    var cnt = 1

    for(first in 0 until n - 1){
        for(third in n - 1 downTo 0){
            if(first + 1 == third)
                break

            if(arr[first] + arr[first + 1] > arr[third]){
                cnt = Math.max(cnt, third - first + 1)
                break
            }
        }
    }

    if(cnt == 1 && n >= 2)
        cnt = 2

    print(cnt)
}